subsys/bluetooth/controller/Kconfig.ll_sw_split

```diff
@@ -76,6 +76,18 @@ config BT_LLL_VENDOR_OPENISA
 	help
 	  Use OpenISA Lower Link Layer implementation.
 
+config BT_CTLR_LTO_SUPPORT
+	bool
+	depends on ARM && SHARED_INTERRUPTS && ISR_TABLES_LOCAL_DECLARATION
+	default y
+	help
+	  Enable use of Link Time Optimizations, to reduce code space usage.
+	  This disables compiler optimization for execution speed or space, and
+	  the Link Time Optimizations apply to the whole of the application.
+
+config LTO
+	select BT_CTLR_LTO_SUPPORT
```

Comment:
Isn't this missing `BT_CTLR` prefix, or are you intentionally overwriting the Kconfig option?

Reply:
This is intentionally appending selection of Controller's LTO support when application enables `CONFIG_LTO=y`. Thereafter hidden `BT_CTLR_LTO_SUPPORT` with its `depends on` will warn the application to include `SHARED_INTERRUPTS`  et al needed as the dependencies for `CONFIG_LTO` support in Controller.

This change is in response discussion with @tejlmand 

```
[9:36 AM] Rasmussen, Torsten
We have an LTO symbol depending on ISR_TABLES_LOCAL_DECLARATION, and now you add another LTOsymbol which aim is to select ISR_TABLES_LOCAL_DECLARATION. Meaning there are two symbols for enabling LTO.
[9:37 AM] Rasmussen, Torsten
[select recommendations](https://docs.zephyrproject.org/latest/build/kconfig/tips.html#id9)[](https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-recommendations)
In summary, here are some recommended practices for select:
Avoid selecting symbols with prompts or dependencies. Prefer depends on. If depends on causes annoying bloat in configuration files, consider adding a Kconfig default for the most common value.
 
 
Kconfig - Tips and Best Practices — Zephyr Project Documentation
[9:37 AM] Rasmussen, Torsten
https://docs.zephyrproject.org/latest/build/kconfig/tips.html#select-recommendations
Kconfig - Tips and Best Practices — Zephyr Project Documentation
[9:38 AM] Rasmussen, Torsten
[Using select for helper symbols](https://docs.zephyrproject.org/latest/build/kconfig/tips.html#id8)[](https://docs.zephyrproject.org/latest/build/kconfig/tips.html#using-select-for-helper-symbols)
A good and safe use of select is for setting “helper” symbols that capture some condition. Such helper symbols should preferably have no prompt or dependencies.
 
 
Kconfig - Tips and Best Practices — Zephyr Project Documentation
```

Reply:
Hmm, so you want `CONFIG_LTO` to enable `CONFIG_BT_CTLR_LTO_SUPPORT` if dependencies are enabled, and else give a warning? 

I think adding 
```c
config LTO
	select BT_CTLR_LTO_SUPPORT
```
is even worse than the `select` that @tejlmand didn't like. 

IMO it should be
```
config BT_CTLR_LTO_SUPPORT
	bool
	depends on LTO && ARM && SHARED_INTERRUPTS && ISR_TABLES_LOCAL_DECLARATION
```
instead.

Reply:
Who selects `BT_CTLR_LTO_SUPPORT` ?

Reply:
> Who selects `BT_CTLR_LTO_SUPPORT` ? i will have to try and get back

What do you mean? I was quoting the code you are adding here

Reply:
Indeed, and I see how that is an issue. 

But it should also be possible to enable `CONFIG_LTO` without getting warnings for `CONFIG_BT_CTLR_LTO_SUPPORT` IMO. 

There may be some Kconfig-magic we can apply to get a good result, but I don't have a good solution right now. I'd like to get some additional eyes on this, just in case that I'm just in the wrong :)